storybookjs/storybook
# Core: Make Storybook esbuild-compatible

This would be a good fundament for adding Snowpack/Vite
builder support. Currently, Storybook UI can be built by
Webpack or tsc, but not esbuild. Esbuild transpiles .ts file-by-file
and therefore has some  caveats:

https://esbuild.github.io/content-types/#typescript-caveats

## What I did

This change adds isolatedModules: true to tsconfig.json,
and fixes a few places in the Storybook source code.
(Mostly `export type`)

The result has been briefly tested with esbuild - with
these changes, it's possible to build Storybook UI
using Vite.

## How to test

I tested building it manually using Esbuild, but my setup is
quite cumbersome to replicate. At least I don't think this change
will break anything - if anything breaks, the automated test
suite should uncover it.
